src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -315,23 +317,15 @@ private function isDuplicateKeyException(WriteException $e): bool
         return 11000 === $code;
     }
 
-    private function getDatabaseVersion(): string
+    private function getReadPreference(): ReadPreference
     {
-        if (null !== $this->databaseVersion) {
-            return $this->databaseVersion;
+        if (null !== $this->readPreference) {
+            return $this->readPreference;
         }
 
-        $command = new Command([
-            'buildinfo' => 1,
-        ]);
-        $cursor = $this->getCollection()->getManager()->executeReadCommand(
-            $this->getCollection()->getDatabaseName(),
-            $command
-        );
-        $buildInfo = $cursor->toArray()[0];
-        $this->databaseVersion = $buildInfo->version;
+        $this->readPreference = new ReadPreference(ReadPreference::RP_PRIMARY);
```

Comment:
Note: `ReadPreference` can be [constructed with a string](https://www.php.net/manual/en/mongodb-driver-readpreference.construct.php) since driver version [1.3.0](https://github.com/mongodb/mongo-php-driver/releases/tag/1.3.0) (September 2017). I didn't find any reference to the `mongodb` extension or PHP library in `composer.json` so perhaps there's no minimum requirement and that's the reason for preferring the deprecated numeric constants (we hope to remove these in 2.0, per [PHPC-1021](https://jira.mongodb.org/browse/PHPC-1021)).

Reply:
symfony doesn't require `ext-mongodb`, the closest it comes is `"require-dev": { "mongodb/mongodb": "~1.1" }`

since `"mongodb/mongodb": "1.1.0"` requires `ext-mongodb: ^1.2.0` i think we should probably support pre `ext-mongodb: 1.3.0` (needs to support ints).

I've added detection for string const vs int const to allow both forwards and backwards compatibility with the shift from `int` to `string` constructed `ReadPreference`

I'm assuming in https://jira.mongodb.org/browse/PHPC-1021 you're planning to just remove `ReadPreference::RP_PRIMARY` and `ReadPreference::PRIMARY` will remain in `ext-mongodb: 2.0`?

Reply:
> symfony doesn't require ext-mongodb, the closest it comes is "require-dev": { "mongodb/mongodb": "~1.1" }

Where is this? I didn't find any reference to `mongodb` in https://github.com/symfony/symfony/blob/5.1/composer.json

> I'm assuming in https://jira.mongodb.org/browse/PHPC-1021 you're planning to just remove ReadPreference::RP_PRIMARY and ReadPreference::PRIMARY will remain in ext-mongodb: 2.0?

Correct. The new constants would stay. The old, integer constants happen to be tied to internal libmongoc constants, which is why we want to remove them (and why the values seemingly make no sense, since they're bitmasks IIRC).

In any event, I think the ternary you have in the current diff is future-proof. :+1:

Reply:
> Where is this? I didn't find any reference to mongodb in https://github.com/symfony/symfony/blob/5.1/composer.json

in the `Component\Lock` - `src/Symfony/Component/Lock/composer.json`